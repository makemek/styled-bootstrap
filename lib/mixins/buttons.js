import { css } from 'styled-components'
import { darken } from 'polished'

export const button_variant = (color, background, border) => css`
  color: ${color};
  background-color: ${background};
  border-color: ${border};

  &:focus,
  &.focus {
    color: ${color};
    background-color: ${darken(.10, background)};
        border-color: ${darken(.25, border)};
  }
  &:hover {
    color: ${color};
    background-color: ${darken(.10, background)};
        border-color: ${darken(.12, border)};
  }
  &:active,
  &.active,
  .open > &.dropdown-toggle {
    color: ${color};
    background-color: ${darken(.10, background)};
        border-color: ${darken(.12, border)};

    &:hover,
    &:focus,
    &.focus {
      color: ${color};
      background-color: ${darken(.17, background)};
          border-color: ${darken(.25, border)};
    }
  }
  &:active,
  &.active,
  .open > &.dropdown-toggle {
    background-image: none;
  }
  &.disabled,
  &[disabled],
  fieldset[disabled] & {
    &:hover,
    &:focus,
    &.focus {
      background-color: ${background};
          border-color: ${border};
    }
  }

  .badge {
    color: ${background};
    background-color: ${color};
  }
`

export const button_size = (padding_vertical, padding_horizontal, font_size, line_height, border_radius) => css`
  padding: ${padding_vertical} ${padding_horizontal};
  font-size: ${font_size};
  line-height: ${line_height};
  border-radius: ${border_radius};
`
