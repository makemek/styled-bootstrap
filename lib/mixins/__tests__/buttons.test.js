import { shallow } from 'enzyme'
import * as buttons from '../buttons'
import styled from 'styled-components'
import React from 'react'
import 'jest-styled-components'

describe('mixin: buttons', () => {

  describe('button_variant', () => {
    it('should match snapshot', () => {
      const Comp = styled.div`
      ${ buttons.button_variant('red', '#ff0000', 'rgb(255,0,0)') }
      `
      const wrapper = shallow(<Comp />)
      expect(wrapper).toMatchSnapshot()
    })
  })

  describe('button_size', () => {
    it('should match snapshot', () => {
      const Comp = styled.div`
      ${ buttons.button_size('10px', '10px', '10px', '1.5', '10px') }
      `
      const wrapper = shallow(<Comp />)
      expect(wrapper).toMatchSnapshot()
    })
  })
})
