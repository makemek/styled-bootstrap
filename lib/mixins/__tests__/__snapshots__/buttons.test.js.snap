// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mixin: buttons button_size should match snapshot 1`] = `
.c0 {
  padding: 10px 10px;
  font-size: 10px;
  line-height: 1.5;
  border-radius: 10px;
}

<div
  className="c0"
/>
`;

exports[`mixin: buttons button_variant should match snapshot 1`] = `
.c0 {
  color: red;
  background-color: #ff0000;
  border-color: rgb(255,0,0);
}

.c0:focus,
.c0.focus {
  color: red;
  background-color: #c00;
  border-color: #800000;
}

.c0:hover {
  color: red;
  background-color: #c00;
  border-color: #c20000;
}

.c0:active,
.c0.active,
.open > .c0.dropdown-toggle {
  color: red;
  background-color: #c00;
  border-color: #c20000;
}

.c0:active:hover,
.c0.active:hover,
.open > .c0.dropdown-toggle:hover,
.c0:active:focus,
.c0.active:focus,
.open > .c0.dropdown-toggle:focus,
.c0:active.focus,
.c0.active.focus,
.open > .c0.dropdown-toggle.focus {
  color: red;
  background-color: #a80000;
  border-color: #800000;
}

.c0:active,
.c0.active,
.open > .c0.dropdown-toggle {
  background-image: none;
}

.c0.disabled:hover,
.c0[disabled]:hover,
fieldset[disabled] .c0:hover,
.c0.disabled:focus,
.c0[disabled]:focus,
fieldset[disabled] .c0:focus,
.c0.disabled.focus,
.c0[disabled].focus,
fieldset[disabled] .c0.focus {
  background-color: #ff0000;
  border-color: rgb(255,0,0);
}

.c0 .badge {
  color: #ff0000;
  background-color: red;
}

<div
  className="c0"
/>
`;
